pkgs/build-support/rust/hooks/cargo-build-hook.sh

```diff
@@ -5,6 +5,10 @@ cargoBuildHook() {
 
     runHook preBuild
 
+    # Let stdenv handle stripping, for consistency and to not break
+    # separateDebugInfo.
+    export "CARGO_PROFILE_${cargoBuildType@U}_STRIP"=false
```

Comment:
Is forcing the build type to uppercase correct? I can't find documentation confirming one way or the other.

Reply:
It's documented [here](https://doc.rust-lang.org/cargo/reference/config.html#environment-variables)